src/triangulum/views.clj

```diff
@@ -0,0 +1,234 @@
+  "Produces the head tag of the index page"
+  [{:keys [bundle-js-files bundle-css-files lang]}]
+  (let [{:keys [title description keywords extra-head-tags gtag-id static-css-files static-js-files]} (get-config :app)]
+    [:head
+     [:title (get title lang "")]
+     [:meta {:charset "utf-8"}]
+     [:meta {:name "description" :content (get description lang "")}]
+     [:meta {:name "keywords" :content (get keywords lang "")}]
+     (seq extra-head-tags)
+     (when gtag-id
+       (list [:script {:async true :src (str "https://www.googletagmanager.com/gtag/js?id=" gtag-id)}]
+             [:script (str "window.dataLayer = window.dataLayer || [];"
+                           "function gtag() { dataLayer.push(arguments); }"
+                           "gtag('js', new Date());"
+                           "gtag('config', '" gtag-id "', {'page_location': location.host + location.pathname});")]))
+     (apply include-css
+            (concat static-css-files
+                    bundle-css-files))
+     (apply include-js static-js-files)
+     (cond
+       ;; CLJS app
+       (cljs-project?)
+       (include-js (find-cljs-app-js))
+
+       ;; JS app/prod
+       (= "prod" (get-config :server :mode))
+       (map (fn [f] [:script {:type "module" :src f}])
+            (butlast bundle-js-files))
+
+       ;; JS app/dev
+       :else
+       (list
+        [:script {:type "module"}
+         "// import RefreshRuntime from 'http://localhost:5173/@react-refresh'
+         // RefreshRuntime.injectIntoGlobalHook(window)
+         window.$RefreshReg$ = () => {}
+         window.$RefreshSig$ = () => (type) => type
+         window.__vite_plugin_react_preamble_installed__ = true"]
+        [:script {:type "module" :src "http://localhost:5173/@vite/client"}]
+        (map (fn [f] [:script {:type "module" :src (str "http://localhost:5173" f)}])
+             (butlast bundle-js-files))))]))
+
+(defn uri->page
+  "Returns the JavaScript file home page"
+  [uri]
+  (->> (str/split uri #"/")
+       (remove str/blank?)
+       (first)
+       ((fnil kebab->camel "home"))))
+
+(defn client-init
+  "Returns the script tag necessary to for the browser to load the app"
+  [entry-file params]
+  (let [js-params (-> params
+                      (assoc :versionDeployed (current-version))
+                      (merge (get-config :app :client-keys))
+                      (json/write-str))]
+    (if-let [cljs-init (get-config :app :cljs-init)]
+      ;; CLJS app
+      [:script {:type "text/javascript"}
+       (format "window.onload = function () { %s(%s); };" (-> cljs-init name kebab->snake) js-params)]
+      ;; JS app
+      [:script {:type "module"}
+       (format "import { pageInit } from \"%s\"; window.onload = function () { pageInit(%s); };"
+               (if (= "prod" (get-config :server :mode))
+                 entry-file
+                 (str "http://localhost:5173" entry-file))
+               js-params)])))
+
+(defn- announcement-banner []
```

Comment:
Do we want to keep this on the backend? In Pyrecast we've already done some work to move it to the frontend. See this file: https://github.com/pyregence/pyregence/blob/main/src/cljs/pyregence/components/page_layout.cljs

Perhaps we should open another ticket for this.